File: Iteratia1/ExtensieRazboi/src/Deck.java
import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class Deck {
    private List<Card> deck;
    private int cardsLeft;

    public Deck(){
        deck=new ArrayList<>();
        for(int i=2;i<=14;i++){
            for(int j=1;j<=4;j++){
                deck.add(new Card(i,j));
            }
        }
        cardsLeft=52;
    }

    public int getNumber(int index){
        return deck.get(index).getNumber();
    }

    public Deck(List<Card> cardList){
        deck=new ArrayList<>();
        deck.addAll(cardList);
        cardsLeft=cardList.size();
    }

    public void addCard(Card carte){
        deck.add(carte);
        cardsLeft++;
    }

    public Deck(String emtpy){
        deck=new ArrayList<>();
        cardsLeft=0;
    }

    public List<Card> getDeck() {
        return deck;
    }

    public void setDeck(List<Card> deck) {
        this.deck = deck;
        cardsLeft=deck.size();
    }
    public void shuffleDeck(){
        int[] choosed=new int[53];
        Random rnd=new Random();
        int numberofCards,position;
        for(int i=0;i<50;i++){
            numberofCards=rnd.nextInt(this.cardsLeft);
            position=rnd.nextInt(this.cardsLeft);
            if(position%4==0){position=52; }
            if(position<numberofCards)position=numberofCards+rnd.nextInt(this.cardsLeft-numberofCards);
            rotateCards(numberofCards,position);
        }
    }
    public void printDeck(){
        System.out.println(deck);
    }

    public int getCardsLeft() {
        return cardsLeft;
    }

    public void setCardsLeft(int cardsLeft) {
        this.cardsLeft = cardsLeft;
    }

    private void rotateCards(int n, int pos){
        List<Card> shufled=new ArrayList<>();

        for(int i=n;i<pos;i++){
            shufled.add(deck.get(i));
        }
        for(int i=0;i<n;i++){
            shufled.add(deck.get(i));
        }
        for(int i=pos;i<this.cardsLeft;i++){
            shufled.add(deck.get(i));
        }
        setDeck(shufled);

    }


    public Card extractOneCard(){
        if(this.deck.size()>0){
            Card card=deck.get(0);
            deck.remove(0);
            this.cardsLeft--;
            return card;
        }else{
            System.out.println("Deck empty");
        }
        return null;
    }

    public List<Card> extractCards(int number){
        List<Card> cards=new ArrayList<>();
        if(number>cardsLeft)number=cardsLeft;
        for (int i = 0; i < number; i++) {
            cards.add(deck.get(0));
            deck.remove(0);
            cardsLeft--;
        }
        return cards;
    }

    public void remove(int index){
        deck.remove(index);
    }

    public Card get(int index){
        return deck.get(index);
    }
}

File: ProjectIP/ProjectIP/Deck.cs
using System;
using System.Collections.Generic;
using System.Globalization;
using System.Text;

namespace ProjectIP
{
    abstract class Deck
    {
        protected List<Card> deck;//primeste o lista de carti
        protected int numberOfCards;// nr de carti pe care le primeste


        public Deck()//constructorul pt deck ul gol
        {
            numberOfCards = 0;
            deck = new List<Card>();
        }


        public void addCard(Card card)
        {
            deck.Add(card);
            numberOfCards++;
        }
        public List<Card> getDeck()
        {
            return deck;
        }
        public void setDeck(List<Card> card)
        {
            deck = card;
            numberOfCards = card.Count;

        }
        public void setNumberOfCards(int numberOfCards)
        { this.numberOfCards = numberOfCards; }
        public int getNumberOfCards()
        { return this.numberOfCards; }
        public void setDeck(Deck deck2)
        {
            this.deck = deck2.deck;
            this.numberOfCards = deck2.numberOfCards;
        }

        public void shuffleDeck()
        {
            var random = new Random();
            int nrCardShuffled, position;
            for (int i = 0; i < 50; i++)
            {

                nrCardShuffled = random.Next(1, numberOfCards - 1);
                position = random.Next(nrCardShuffled, numberOfCards);
                if (position % 5 == 0) { position = numberOfCards; }
                rotateCards(nrCardShuffled, position);
            }
        }
        private void rotateCards(int nrCards, int pos)
        {
            List<Card> shufled = new List<Card>();
            int i;
            for (i = nrCards; i < pos; i++)
            {
                shufled.Add(deck[i]);
            }
            for (i = 0; i < nrCards; i++)
            {
                shufled.Add(deck[i]);
            }
            for (i = pos; i < this.numberOfCards; i++)
            {
                shufled.Add(deck[i]);
            }
            setDeck(shufled);
        }
        public void printDeck()
        {
            foreach (Card card in deck)
            {
                Console.Write(card + ",");
            }
            Console.WriteLine();
        }
        public Card extractOneCard()
        {
            Card card = deck[0];
            deck.RemoveAt(0);
            return card;

        }
        public List<Card> extractNCards(int n)
        {
            List<Card> cards = new List<Card>();
            if (n > numberOfCards) n = numberOfCards;
            for(int i = 0; i < n; i++)
            {
                cards.Add(deck[0]);
                deck.RemoveAt(0);
            }
            return cards;
        }

    }
}
